src/js/actions/layers.js

```diff
@@ -1377,45 +1382,154 @@ define(function (require, exports) {
+            if (currentLayer.bounds.right > selectedLayer.bounds.right) {
+                return currentLayer;
+            } else {
+                return selectedLayer;
+            }
+        }, artboards.first());
+        return layer;
+    };
+
+    /**
+     * Helper function to get Bounds from specs (template preset)
+     *
+     * @private
+     * @param {object} specs 
+     * @param {Immutable.List.<Layer>} artboards
+     * @param {Layer} layerRef
+     * @return {{finalBounds: Bounds, layerRef: Layer}}
+     */
+    var _getBoundsFromSpecs = function (specs, artboards, layerRef) {
+        var preset = specs.preset,
+            width,
+            height,
+            finalBounds;
+
+        var index = _.findIndex(templates, function (obj) {
+                return obj.preset === preset;
+            });
+
+        height = templates[index].height;
+        width = templates[index].width;
+
+
+        // if artboards are empty, insert template at first position 
+        if (artboards.isEmpty()) {
+            layerRef = layerLib.referenceBy.current;
+            finalBounds = new Bounds({
+                top: DEFAULT_ARTBOARD_BOUNDS.top,
+                bottom: height,
+                left: DEFAULT_ARTBOARD_BOUNDS.left,
+                right: width
+            });
+        } else {
+            // find the rightmost artboard and insert the template after this 
+            var rightMostLayer = _findRightMostArtboard(artboards);
+            var offset = 100;
+            finalBounds = rightMostLayer.bounds.merge({
+                bottom: rightMostLayer.bounds.top + height,
+                left: rightMostLayer.bounds.left + offset + rightMostLayer.bounds.width,
+                right: rightMostLayer.bounds.right + offset + width
+            });
+        }
+
+        return {
+            finalBounds: finalBounds,
+            layerRef: layerRef
+        };
+    };
+
+    /**
+     * Helper function to get Bounds from specs (template preset)
+     * 
+     * @private
+     * @param {Immutable.List.<Layer>} artboards
+     * @param {Immutable.List.<Layer>} selected 
+     * @param {Layer} layerRef
+     * @return {{finalBounds: Bounds, layerRef: Layer}}
+     */
+
+    var _getBoundsFromNoInput = function (artboards, selectedArtboards, layerRef) {
+        var finalBounds;
+
+        // if there are no artboards in the document, the new artboard should have dimensions of default
+        if (artboards.isEmpty()) {
+            layerRef = layerLib.referenceBy.current;
+            finalBounds = new Bounds({
+                top: DEFAULT_ARTBOARD_BOUNDS.top,
+                bottom: DEFAULT_ARTBOARD_BOUNDS.bottom,
+                left: DEFAULT_ARTBOARD_BOUNDS.left,
+                right: DEFAULT_ARTBOARD_BOUNDS.right
+            });
+        } else {
+            // else we want to get the right most artboard 
+            var rightMostLayer = _findRightMostArtboard(artboards);
+            var offset = 100;
```

Comment:
Do you mean rightMostLayer, offset with the declaration of var finalBounds at the top? I included them down here because I thought in the last review, you said that I should only declare them in the part of the if statement that uses them? Maybe I misunderstood...